### DISCUSSION TITLE
Does next-auth support special characters in client secret?

### DISCUSSION THREAD

**Discussion by bl0up** [nextauthjs/next-auth#5549] [Help]
### Question 💬

Hello, 

I am trying to authenticate on a server using **oauth** and the _authorization code grant_ mechanism. I do not have access to the server configuration, but I know that they are using the php league oauth2 server. 

In my `[...nextauth.js]` file, I defined a custom provider.

On my website, I have a login button that redirects to their authentication page. There, I can enter my user id and user password and if both are correct, I am also authenticated on my website. 

Everything works fine if the secret (process.env.CLIENT_SECRET) that the admin provides does not contain any special characters (e.g: AB20Cde4). However, when special characters are added to the secret (e.g AB20-Cde-4), authentication fails and I receive the following error message: 
<img width="973" alt="Screenshot 2022-10-10 at 12 20 08" src="https://user-images.githubusercontent.com/19263787/194845143-11433321-018d-4b64-a758-d769f4d778af.png">

Is there something I am missing in the `[...nextauth.js]` file to make it work with special characters? 

### How to reproduce ☕️

This is the `[...nextauth.js]` config that I am using.
```typescript
{
    id: 'id',
    name: 'name',
    type: 'oauth',
    scope: 'login',
    version: '2.0',
    clientId: process.env.CLIENT_ID,
    clientSecret: process.env.CLIENT_SECRET,
    authorization: {
        url: process.env.AUTH_URL + '/authorize',
        params: { scope: 'login'},
    },
    token: {
        url: process.env.AUTH_URL + '/access_token',
        params: { grant_type: 'authorization_code' },
    },
    userinfo: process.env.AUTH_URL + '/owner_details',
    async profile(profile) {
        return Promise.resolve({
            id: profile.user_id,
            email: profile.email,
        });
     },
 }
```

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by balazsorban44** [comment]
NextAuth.js has no opinions on the secret value. Remember, the client secret is coming from your IdP, not a random value. Make sure to copy it carefully/correctly. Make also sure that the environment variable is correctly read into the app.
